diff --git a/Makefile b/Makefile
index 44d73f4..48be680 100644
--- a/Makefile
+++ b/Makefile
@@ -145,7 +145,7 @@ COMMON_UNITTEST_CFLAGS += $(CODEC_UNITTEST_CFLAGS)
 .PHONY: test gtest-bootstrap clean $(PROJECT_NAME).pc $(PROJECT_NAME)-static.pc
 
 generate-version:
-	$(QUIET)sh $(SRC_PATH)codec/common/generate_version.sh $(SRC_PATH)
+	$(QUIET)sh $(SRC_PATH)/codec/common/generate_version.sh $(SRC_PATH)
 
 codec/decoder/plus/src/welsDecoderExt.$(OBJ): | generate-version
 codec/encoder/plus/src/welsEncoderExt.$(OBJ): | generate-version
diff --git a/build/gtest-targets.mk b/build/gtest-targets.mk
index fe71196..a3b42d5 100644
--- a/build/gtest-targets.mk
+++ b/build/gtest-targets.mk
@@ -6,7 +6,7 @@ GTEST_OBJS += $(GTEST_CPP_SRCS:.cc=.$(OBJ))
 
 OBJS += $(GTEST_OBJS)
 
-$(GTEST_SRCDIR)/%.$(OBJ): $(GTEST_SRCDIR)/%.cc
+$(GTEST_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(GTEST_SRCDIR)/%.cc
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(GTEST_CFLAGS) $(GTEST_INCLUDES) -c $(CXX_O) $<
 
 $(LIBPREFIX)gtest.$(LIBSUFFIX): $(GTEST_OBJS)
diff --git a/build/msvc-common.mk b/build/msvc-common.mk
index 33d5985..b4e287e 100644
--- a/build/msvc-common.mk
+++ b/build/msvc-common.mk
@@ -41,7 +41,7 @@ SHAREDLIBSUFFIXVER=$(SHAREDLIBSUFFIX)
 SHARED=-LD
 EXTRA_LIBRARY=$(PROJECT_NAME)_dll.lib
 LDFLAGS += -link
-SHLDFLAGS=-pdb:$(PROJECT_NAME).pdb -def:$(SRC_PATH)openh264.def -implib:$(EXTRA_LIBRARY)
+SHLDFLAGS=-pdb:$(PROJECT_NAME).pdb -def:$(SRC_PATH)/openh264.def -implib:$(EXTRA_LIBRARY)
 STATIC_LDFLAGS=
 CODEC_UNITTEST_CFLAGS=-D_CRT_SECURE_NO_WARNINGS
 
diff --git a/codec/common/generate_version.sh b/codec/common/generate_version.sh
index 8814b46..2d5250a 100755
--- a/codec/common/generate_version.sh
+++ b/codec/common/generate_version.sh
@@ -1,4 +1,5 @@
 #!/bin/bash
+SRC_PATH=$1
 git rev-list HEAD | sort > config.git-hash
 SRC_PATH=$1
 LOCALVER=`wc -l config.git-hash | awk '{print $1}'`
diff --git a/codec/common/targets.mk b/codec/common/targets.mk
index a815d19..94e349f 100644
--- a/codec/common/targets.mk
+++ b/codec/common/targets.mk
@@ -61,13 +61,13 @@ OBJS += $(COMMON_OBJSARM64)
 
 OBJS += $(COMMON_OBJS)
 
-$(COMMON_SRCDIR)/%.$(OBJ): $(COMMON_SRCDIR)/%.cpp
+$(COMMON_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(COMMON_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(COMMON_CFLAGS) $(COMMON_INCLUDES) -c $(CXX_O) $<
 
-$(COMMON_SRCDIR)/%.$(OBJ): $(COMMON_SRCDIR)/%.asm
+$(COMMON_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(COMMON_SRCDIR)/%.asm
 	$(QUIET_ASM)$(ASM) $(ASMFLAGS) $(ASM_INCLUDES) $(COMMON_ASMFLAGS) $(COMMON_ASM_INCLUDES) -o $@ $<
 
-$(COMMON_SRCDIR)/%.$(OBJ): $(COMMON_SRCDIR)/%.S
+$(COMMON_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(COMMON_SRCDIR)/%.S
 	$(QUIET_CCAS)$(CCAS) $(CCASFLAGS) $(ASMFLAGS) $(INCLUDES) $(COMMON_CFLAGS) $(COMMON_INCLUDES) -c -o $@ $<
 
 $(LIBPREFIX)common.$(LIBSUFFIX): $(COMMON_OBJS)
diff --git a/codec/console/common/targets.mk b/codec/console/common/targets.mk
index 3065783..ee41435 100644
--- a/codec/console/common/targets.mk
+++ b/codec/console/common/targets.mk
@@ -6,7 +6,7 @@ CONSOLE_COMMON_OBJS += $(CONSOLE_COMMON_CPP_SRCS:.cpp=.$(OBJ))
 
 OBJS += $(CONSOLE_COMMON_OBJS)
 
-$(CONSOLE_COMMON_SRCDIR)/%.$(OBJ): $(CONSOLE_COMMON_SRCDIR)/%.cpp
+$(CONSOLE_COMMON_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(CONSOLE_COMMON_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(CONSOLE_COMMON_CFLAGS) $(CONSOLE_COMMON_INCLUDES) -c $(CXX_O) $<
 
 $(LIBPREFIX)console_common.$(LIBSUFFIX): $(CONSOLE_COMMON_OBJS)
diff --git a/codec/console/dec/targets.mk b/codec/console/dec/targets.mk
index 70051c9..de0f05a 100644
--- a/codec/console/dec/targets.mk
+++ b/codec/console/dec/targets.mk
@@ -7,7 +7,7 @@ H264DEC_OBJS += $(H264DEC_CPP_SRCS:.cpp=.$(OBJ))
 
 OBJS += $(H264DEC_OBJS)
 
-$(H264DEC_SRCDIR)/%.$(OBJ): $(H264DEC_SRCDIR)/%.cpp
+$(H264DEC_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(H264DEC_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(H264DEC_CFLAGS) $(H264DEC_INCLUDES) -c $(CXX_O) $<
 
 h264dec$(EXEEXT): $(H264DEC_OBJS) $(H264DEC_DEPS)
diff --git a/codec/console/enc/targets.mk b/codec/console/enc/targets.mk
index 2400944..4cbda2b 100644
--- a/codec/console/enc/targets.mk
+++ b/codec/console/enc/targets.mk
@@ -6,7 +6,7 @@ H264ENC_OBJS += $(H264ENC_CPP_SRCS:.cpp=.$(OBJ))
 
 OBJS += $(H264ENC_OBJS)
 
-$(H264ENC_SRCDIR)/%.$(OBJ): $(H264ENC_SRCDIR)/%.cpp
+$(H264ENC_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(H264ENC_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(H264ENC_CFLAGS) $(H264ENC_INCLUDES) -c $(CXX_O) $<
 
 h264enc$(EXEEXT): $(H264ENC_OBJS) $(H264ENC_DEPS)
diff --git a/codec/decoder/targets.mk b/codec/decoder/targets.mk
index d45a002..f751350 100644
--- a/codec/decoder/targets.mk
+++ b/codec/decoder/targets.mk
@@ -55,13 +55,13 @@ OBJS += $(DECODER_OBJSARM64)
 
 OBJS += $(DECODER_OBJS)
 
-$(DECODER_SRCDIR)/%.$(OBJ): $(DECODER_SRCDIR)/%.cpp
+$(DECODER_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(DECODER_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(DECODER_CFLAGS) $(DECODER_INCLUDES) -c $(CXX_O) $<
 
-$(DECODER_SRCDIR)/%.$(OBJ): $(DECODER_SRCDIR)/%.asm
+$(DECODER_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(DECODER_SRCDIR)/%.asm
 	$(QUIET_ASM)$(ASM) $(ASMFLAGS) $(ASM_INCLUDES) $(DECODER_ASMFLAGS) $(DECODER_ASM_INCLUDES) -o $@ $<
 
-$(DECODER_SRCDIR)/%.$(OBJ): $(DECODER_SRCDIR)/%.S
+$(DECODER_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(DECODER_SRCDIR)/%.S
 	$(QUIET_CCAS)$(CCAS) $(CCASFLAGS) $(ASMFLAGS) $(INCLUDES) $(DECODER_CFLAGS) $(DECODER_INCLUDES) -c -o $@ $<
 
 $(LIBPREFIX)decoder.$(LIBSUFFIX): $(DECODER_OBJS)
diff --git a/codec/encoder/targets.mk b/codec/encoder/targets.mk
index b3e5652..dab7d8a 100644
--- a/codec/encoder/targets.mk
+++ b/codec/encoder/targets.mk
@@ -78,13 +78,13 @@ OBJS += $(ENCODER_OBJSARM64)
 
 OBJS += $(ENCODER_OBJS)
 
-$(ENCODER_SRCDIR)/%.$(OBJ): $(ENCODER_SRCDIR)/%.cpp
+$(ENCODER_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(ENCODER_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(ENCODER_CFLAGS) $(ENCODER_INCLUDES) -c $(CXX_O) $<
 
-$(ENCODER_SRCDIR)/%.$(OBJ): $(ENCODER_SRCDIR)/%.asm
+$(ENCODER_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(ENCODER_SRCDIR)/%.asm
 	$(QUIET_ASM)$(ASM) $(ASMFLAGS) $(ASM_INCLUDES) $(ENCODER_ASMFLAGS) $(ENCODER_ASM_INCLUDES) -o $@ $<
 
-$(ENCODER_SRCDIR)/%.$(OBJ): $(ENCODER_SRCDIR)/%.S
+$(ENCODER_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(ENCODER_SRCDIR)/%.S
 	$(QUIET_CCAS)$(CCAS) $(CCASFLAGS) $(ASMFLAGS) $(INCLUDES) $(ENCODER_CFLAGS) $(ENCODER_INCLUDES) -c -o $@ $<
 
 $(LIBPREFIX)encoder.$(LIBSUFFIX): $(ENCODER_OBJS)
diff --git a/codec/processing/targets.mk b/codec/processing/targets.mk
index b14a632..4cb0639 100644
--- a/codec/processing/targets.mk
+++ b/codec/processing/targets.mk
@@ -57,13 +57,13 @@ OBJS += $(PROCESSING_OBJSARM64)
 
 OBJS += $(PROCESSING_OBJS)
 
-$(PROCESSING_SRCDIR)/%.$(OBJ): $(PROCESSING_SRCDIR)/%.cpp
+$(PROCESSING_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(PROCESSING_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(PROCESSING_CFLAGS) $(PROCESSING_INCLUDES) -c $(CXX_O) $<
 
-$(PROCESSING_SRCDIR)/%.$(OBJ): $(PROCESSING_SRCDIR)/%.asm
+$(PROCESSING_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(PROCESSING_SRCDIR)/%.asm
 	$(QUIET_ASM)$(ASM) $(ASMFLAGS) $(ASM_INCLUDES) $(PROCESSING_ASMFLAGS) $(PROCESSING_ASM_INCLUDES) -o $@ $<
 
-$(PROCESSING_SRCDIR)/%.$(OBJ): $(PROCESSING_SRCDIR)/%.S
+$(PROCESSING_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(PROCESSING_SRCDIR)/%.S
 	$(QUIET_CCAS)$(CCAS) $(CCASFLAGS) $(ASMFLAGS) $(INCLUDES) $(PROCESSING_CFLAGS) $(PROCESSING_INCLUDES) -c -o $@ $<
 
 $(LIBPREFIX)processing.$(LIBSUFFIX): $(PROCESSING_OBJS)
diff --git a/module/targets.mk b/module/targets.mk
index f80bd7a..d322eb5 100644
--- a/module/targets.mk
+++ b/module/targets.mk
@@ -6,6 +6,6 @@ MODULE_OBJS += $(MODULE_CPP_SRCS:.cpp=.$(OBJ))
 
 OBJS += $(MODULE_OBJS)
 
-$(MODULE_SRCDIR)/%.$(OBJ): $(MODULE_SRCDIR)/%.cpp
+$(MODULE_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(MODULE_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(MODULE_CFLAGS) $(MODULE_INCLUDES) -c $(CXX_O) $<
 
diff --git a/test/api/targets.mk b/test/api/targets.mk
index 093969b..6a35d40 100644
--- a/test/api/targets.mk
+++ b/test/api/targets.mk
@@ -19,9 +19,9 @@ API_TEST_C_SRCS=\
 API_TEST_OBJS += $(API_TEST_C_SRCS:.c=.$(OBJ))
 OBJS += $(API_TEST_OBJS)
 
-$(API_TEST_SRCDIR)/%.$(OBJ): $(API_TEST_SRCDIR)/%.cpp
+$(API_TEST_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(API_TEST_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(API_TEST_CFLAGS) $(API_TEST_INCLUDES) -c $(CXX_O) $<
 
-$(API_TEST_SRCDIR)/%.$(OBJ): $(API_TEST_SRCDIR)/%.c
+$(API_TEST_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(API_TEST_SRCDIR)/%.c
 	$(QUIET_CC)$(CC) $(CFLAGS) $(INCLUDES) $(API_TEST_CFLAGS) $(API_TEST_INCLUDES) -c $(CXX_O) $<
 
diff --git a/test/common/targets.mk b/test/common/targets.mk
index ad07d1c..23960cb 100644
--- a/test/common/targets.mk
+++ b/test/common/targets.mk
@@ -6,6 +6,6 @@ COMMON_UNITTEST_OBJS += $(COMMON_UNITTEST_CPP_SRCS:.cpp=.$(OBJ))
 
 OBJS += $(COMMON_UNITTEST_OBJS)
 
-$(COMMON_UNITTEST_SRCDIR)/%.$(OBJ): $(COMMON_UNITTEST_SRCDIR)/%.cpp
+$(COMMON_UNITTEST_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(COMMON_UNITTEST_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(COMMON_UNITTEST_CFLAGS) $(COMMON_UNITTEST_INCLUDES) -c $(CXX_O) $<
 
diff --git a/test/decoder/targets.mk b/test/decoder/targets.mk
index 2233989..ccf6678 100644
--- a/test/decoder/targets.mk
+++ b/test/decoder/targets.mk
@@ -13,6 +13,6 @@ DECODER_UNITTEST_OBJS += $(DECODER_UNITTEST_CPP_SRCS:.cpp=.$(OBJ))
 
 OBJS += $(DECODER_UNITTEST_OBJS)
 
-$(DECODER_UNITTEST_SRCDIR)/%.$(OBJ): $(DECODER_UNITTEST_SRCDIR)/%.cpp
+$(DECODER_UNITTEST_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(DECODER_UNITTEST_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(DECODER_UNITTEST_CFLAGS) $(DECODER_UNITTEST_INCLUDES) -c $(CXX_O) $<
 
diff --git a/test/encoder/targets.mk b/test/encoder/targets.mk
index 8727955..e8aefb5 100644
--- a/test/encoder/targets.mk
+++ b/test/encoder/targets.mk
@@ -21,6 +21,6 @@ ENCODER_UNITTEST_OBJS += $(ENCODER_UNITTEST_CPP_SRCS:.cpp=.$(OBJ))
 
 OBJS += $(ENCODER_UNITTEST_OBJS)
 
-$(ENCODER_UNITTEST_SRCDIR)/%.$(OBJ): $(ENCODER_UNITTEST_SRCDIR)/%.cpp
+$(ENCODER_UNITTEST_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(ENCODER_UNITTEST_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(ENCODER_UNITTEST_CFLAGS) $(ENCODER_UNITTEST_INCLUDES) -c $(CXX_O) $<
 
diff --git a/test/processing/targets.mk b/test/processing/targets.mk
index 2ce49a2..8f0bacd 100644
--- a/test/processing/targets.mk
+++ b/test/processing/targets.mk
@@ -9,6 +9,6 @@ PROCESSING_UNITTEST_OBJS += $(PROCESSING_UNITTEST_CPP_SRCS:.cpp=.$(OBJ))
 
 OBJS += $(PROCESSING_UNITTEST_OBJS)
 
-$(PROCESSING_UNITTEST_SRCDIR)/%.$(OBJ): $(PROCESSING_UNITTEST_SRCDIR)/%.cpp
+$(PROCESSING_UNITTEST_SRCDIR)/%.$(OBJ): $(SRC_PATH)/$(PROCESSING_UNITTEST_SRCDIR)/%.cpp
 	$(QUIET_CXX)$(CXX) $(CFLAGS) $(CXXFLAGS) $(INCLUDES) $(PROCESSING_UNITTEST_CFLAGS) $(PROCESSING_UNITTEST_INCLUDES) -c $(CXX_O) $<
 
